packages/web/src/algo-signer-types.ts

```diff
@@ -0,0 +1,136 @@
+  readonly threshold: number
+  readonly addrs: string[]
+}
+
+export interface WalletTransaction {
+  readonly txn: string
+  readonly signers?: string[]
+  readonly message?: string
+  readonly msig?: WalletMultisigMetadata
+  readonly authAddr?: string
+}
+
+export const JSONRPC_VERSION: string = '2.0';
+
+/* eslint-disable no-unused-vars */
+export enum JsonRpcMethod {
+  Heartbeat = 'heartbeat',
+  Authorization = 'authorization',
+  AuthorizationAllow = 'authorization-allow',
+  AuthorizationDeny = 'authorization-deny',
+  SignAllow = 'sign-allow',
+  SignAllowMultisig = 'sign-allow-multisig',
+  SignAllowWalletTx = 'sign-allow-wallet-tx',
+  SignDeny = 'sign-deny',
+  SignTransaction = 'sign-transaction',
+  SignMultisigTransaction = 'sign-multisig-transaction',
+  SignWalletTransaction = 'sign-wallet-transaction',
+  SendTransaction = 'send-transaction',
+  Algod = 'algod',
+  Indexer = 'indexer',
+  Accounts = 'accounts',
+  // UI methods
+  CreateWallet = 'create-wallet',
+  DeleteWallet = 'delete-wallet',
+  CreateAccount = 'create-account',
+  SaveAccount = 'save-account',
+  ImportAccount = 'import-account',
+  DeleteAccount = 'delete-account',
+  GetSession = 'get-session',
+  Login = 'login',
+  Logout = 'logout',
+  AccountDetails = 'account-details',
+  Transactions = 'transactions',
+  AssetDetails = 'asset-details',
+  AssetsAPIList = 'assets-api-list',
+  AssetsVerifiedList = 'assets-verified-list',
+  SignSendTransaction = 'sign-send-transaction',
+  ChangeLedger = 'change-ledger',
+  SaveNetwork = 'save-network',
+  DeleteNetwork = 'delete-network',
+  GetLedgers = 'get-ledgers',
+}
+
+export interface JsonPayload {
+  [key: string]: string | number | WalletTransaction[] | JsonPayload | undefined
+}
+
+export interface JsonRpcBody {
+  readonly jsonrpc: string
+  readonly method: JsonRpcMethod
+  readonly params: JsonPayload
+  readonly id: string
+}
+
+export enum MessageSource {
+  Extension = 'extension',
+  DApp = 'dapp',
+  Router = 'router',
+  UI = 'ui',
+}
+export interface MessageBody {
+  readonly source: MessageSource
+  readonly body: JsonRpcBody
+}
+
+export type JsonRpcResponse = string;
+
+export interface Encoding {
+  msgpackToBase64: (txn: Uint8Array) => string
+
+  base64ToMsgpack: (txn: string) => Uint8Array
+}
+
+export interface AlgoSigner {
+  encoding: Encoding
+
+  accounts: (params: JsonPayload, error?: RequestErrors) => Promise<JsonPayload>
+
+  sign: (params: Transaction, error?: RequestErrors) => Promise<JsonPayload>
+
+  signMultisig: (
+    params: MultisigTransaction,
+    error?: RequestErrors
+  ) => Promise<JsonPayload>
+
+  send: (params: any, error?: RequestErrors) => Promise<JsonPayload>
+
+  algod: (params: JsonPayload, error?: RequestErrors) => Promise<JsonPayload>
+
+  indexer: (params: JsonPayload, error?: RequestErrors) => Promise<JsonPayload>
+
+  signTxn: (
+    transactions: WalletTransaction[],
+    error?: RequestErrors
+  ) => Promise<JsonPayload>
+}
```

Comment:
So these types are not exported in algosigner? Can we request them to export these typings and add a TODO here?
I think that will be good because then we won't need to maintain their types.

Reply:
I thought that but How will we import those ? Algosigner is a chrome extension, we cannot import it, it is not present in npm, I couldn't find a way to import extension in package,
Please let me know if you find something.

Reply:
hmm, makes sense. I think we can ask once about this on their discord, maybe they'll release their typings under `@types/algosigner`.

Reply:
ok, will ask